pkg/config/schedule/schedule.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Allstar Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package schedule provides the ActionSchedule type for use in config and
+// when deciding whether to perform an action.
+package schedule
+
+import (
+	"strings"
+	"time"
+)
+
+type ScheduleAction string
+
+const (
+	ScheduleActionLog         ScheduleAction = "log"
+	ScheduleActionIssueCreate ScheduleAction = "issue"
+	ScheduleActionIssuePing   ScheduleAction = "ping"
+	ScheduleActionFix         ScheduleAction = "fix"
+)
+
+// actionOverrides specifies, for each action A, a set of actions B.
+// If any action in B is disabled, action A will be disabled.
+var actionOverrides = map[ScheduleAction][]ScheduleAction{
+	ScheduleActionIssuePing: {ScheduleActionIssueCreate},
+}
+
+type ScheduleConfig struct {
```

Comment:
Move this struct to the config package. I'd like all the structs that define what goes into `allstar.yaml` in one place.